ca/ca.go

```diff
@@ -160,16 +160,16 @@ func (a *Ca) PktTox509(pktCom []byte, caBytes []byte) ([]byte, error) {
 	}
 
 	var payload struct {
-		Issuer   string `json:"iss"`
-		Audience string `json:"aud"`
-		Email    string `json:"email"`
+		Issuer   string   `json:"iss"`
+		Audience []string `json:"aud"`
```

Comment:
Why is this changing?

Reply:
its a bug fix, not sure how it worked before (failing tests) and audience is a list.
[oidc spec](https://openid.net/specs/openid-connect-core-1_0.html):
>aud
REQUIRED. Audience(s) that this ID Token is intended for. It MUST contain the OAuth 2.0 client_id of the Relying Party as an audience value. It MAY also contain identifiers for other audiences. In the general case, the aud value is an array of case-sensitive strings. In the common special case when there is one audience, the aud value MAY be a single case-sensitive string.

Reply:
Audience as defined in the oidc spec can be a JSON list, but most OPs I've come across don't represent this as list when it only contains one element. 

Our mock PK Token always treats aud as a JSON list. Our OPs always treat it as a string. If you deserialize it as a list, it works for the unittests but breaks for google. If you treat it as a string, it works for google and breaks for the unittests.

This was how I fixed this issue in PR #68 

```golang
// An audience can be a string or an array of strings.
//
// RFC-7519 JSON Web Token (JWT) says:
// "In the general case, the "aud" value is an array of case-
// sensitive strings, each containing a StringOrURI value.  In the
// special case when the JWT has one audience, the "aud" value MAY be a
// single case-sensitive string containing a StringOrURI value."
// https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3
var audience string
switch t := claims.Aud.(type) {
case string:
audience = t
case []any:
audList := []string{}
for _, v := range t {
	audList = append(audList, v.(string))
}
audience = strings.Join(audList, ",")
default:
return nil, fmt.Errorf("failed to deserialize aud (audience) claim in ID Token: %d", t)
}
```
https://github.com/openpubkey/openpubkey/pull/68/files#diff-56229f53c5a7b115aa512c3be0077f3e33c4b7b91bc7cc303ecc14f50ddc699bR179-R199

Reply:
I can add a common `claims` struct type with a custom json unmarshalling method to cover both cases like the code you referenced. Agreed, it is annoying to spec a field that can be two different data types.

Reply:
what do you think about this? [e7b3e8a](https://github.com/openpubkey/openpubkey/pull/79/commits/e7b3e8a2e7cdf09c7db10a78e68120d0f11359ef)

Reply:
I think a common claims struct as you have done here would be very helpful in a lot of places. Addressing the "Schneiders Audience" issue where you don't know if something is an list of a list until you deserialize it is something I've been planning to fix each time I add unit tests.

This is a really nice way of fixing it and could simplify a lot of code.

Reply:
Thanks, I had to do something similar recently for unmarshalling the JWKS response while preserving the raw JWK contents. Every OP had a bit different JWK schema, so it was necessary to preserve the original JSON. Especially since some of the JWK claims were not in the standard.